### PR TITLE
Added most

### PR DESCRIPTION
**Pull request by Risto-Stevcev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The solutions are best read in the following order:
   - [__await.js__](./await.js)
   - [__tasks.js__](./tasks.js)
   - [__lazy-either.js__](./lazy-either.js)
+  - [__most.js__](./most.js)
```
Comment (davidchambers) [inline README.md]: Let's group this with the other Promise-based solutions (after __promises-ramda.js__, say).

File: most.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const S = require('sanctuary');
+
+
+//    join :: String -> String -> String
+const join = R.curryN(2, path.join);
+
+//    readFile :: String -> String -> Promise Error String
+const readFile = R.curry((encoding, filename) =>
+  new Promise((resolve, reject) =>
+    fs.readFile(filename, { encoding }, (err, data) =>
```
Comment (davidchambers) [inline most.js]: Let's use `{encoding: encoding}` here. I'd like to keep the code compatible with Node 0.12.x.

File: most.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    )));
+
+//    concatFiles :: String -> Promise Error String
+const concatFiles = dir =>
+  R.pipe(R.pipe(readFile('utf8'), most.fromPromise),
+         R.map(S.lines),
+         R.chain(R.pipe(R.map(join(dir)), most.from)),
+         R.map(readFile('utf8')),
+         stream => stream.await(),
+         R.reduce(R.concat, ''))(join(dir, 'index.txt'));
```
Comment (davidchambers) [inline most.js]: At first glance `R.reduce(R.concat, ''))(join(dir, 'index.txt'))` looks like a single expression.

Let's use this formatting:

```javascript
R.pipe(
  ...,
  ...,
  ...,
  ...,
  ...,
  ...
)(join(dir, 'index.txt'))
```

**Comment by davidchambers** [comment]
Thank you, @Risto-Stevcev!

**Comment by Risto-Stevcev** [comment]
Updated

**Pull request merged**